doc/array_buffer.md

```diff
@@ -1,5 +1,127 @@
-# Array buffer
+# ArrayBuffer
 
-You are reading a draft of the next documentation and it's in continuos update so
-if you don't find what you need please refer to:
-[C++ wrapper classes for the ABI-stable C APIs for Node.js](https://nodejs.github.io/node-addon-api/)
+The `ArrayBuffer` class corresponds to the JavaScript `ArrayBuffer` class.
+
+## Methods
+
+### New
+
+Allocates a new `ArrayBuffer` object with a given length.
+
+```cpp
+static ArrayBuffer New(napi_env env, size_t byteLength);
+```
+
+- `[in] env`: The environment in which to create the ArrayBuffer object.
+- `[in] byteLength`: The length to be allocated, in bytes.
+
+Returns a new `ArrayBuffer` object.
+
+### New
+
+Wraps the provided external data into a new `ArrayBuffer` object.
+
+The `ArrayBuffer` object does not assume ownership for the data and expects it
+to be valid for the lifetime of the object. Since the `ArrayBuffer` is subject
+to garbage collection this overload is only suitable for data which is static
+and never needs to be freed.
+
+```cpp
+static ArrayBuffer New(napi_env env, void* externalData, size_t byteLength);
+```
+
+- `[in] env`: The environment in which to create the `ArrayBuffer` object.
+- `[in] externalData`: The pointer to the external data to wrap.
+- `[in] byteLength`: The length of the `externalData`, in bytes.
+
+Returns a new `ArrayBuffer` object.
+
+### New
+
+Wraps the provided external data into a new `ArrayBuffer` object.
+
+The `ArrayBuffer` object does not assume ownership for the data and expects it
```

Comment:
It is not more accurate that it effectively does take ownership, taking responsibility to run the finalizer to provided to delete the data when the class is deleted?

Reply:
I toyed with the wording here a bit, but since the module code is still responsible for cleanup I feel like it's more accurate to say that ownership is never transferred.  The caller is merely lending the data to the `ArrayBuffer`.  Doing a bit more reading, it seems to align pretty well with the way V8 describes these "externalized" `ArrayBuffer` objects.

https://v8.paulfryzel.com/docs/master/classv8_1_1_array_buffer.html#a166848a34653afd4502e4cc33443815d

> After ArrayBuffer had been externalized, it does no longer own the memory block. The caller should take steps to free memory when it is no longer needed.

Reply:
@kfarnung  did we close on what we wanted to do?  It would be good to get this landed.